airbyte-server/src/test/java/io/airbyte/server/handlers/ConnectionsHandlerTest.java

```diff
@@ -312,52 +327,270 @@ void testCreateConnectionWithBadDefinitionIds() throws JsonValidationException,
+          standardSync.getSourceCatalogId())
+          .schedule(null)
+          .scheduleType(ConnectionScheduleType.CRON)
+          .scheduleData(cronScheduleData);
+
+      final StandardSync expectedPersistedSync = Jsons.clone(standardSync)
+          .withSchedule(null)
+          .withScheduleType(ScheduleType.CRON)
+          .withScheduleData(new ScheduleData().withCron(new Cron().withCronExpression(CRON_EXPRESSION).withCronTimeZone(CRON_TIMEZONE_UTC)))
+          .withManual(false);
+
+      when(configRepository.getStandardSync(standardSync.getConnectionId())).thenReturn(standardSync);
+
+      final ConnectionRead actualConnectionRead = connectionsHandler.updateConnection(connectionUpdate);
+
+      assertEquals(expectedRead, actualConnectionRead);
+      verify(configRepository).writeStandardSync(expectedPersistedSync);
+      verify(eventRunner).update(connectionUpdate.getConnectionId());
+    }
+
+    @Test
+    void testUpdateConnectionPatchBasicSchedule() throws Exception {
+
+      final ConnectionScheduleData newScheduleData =
+          new ConnectionScheduleData().basicSchedule(new ConnectionScheduleDataBasicSchedule().timeUnit(TimeUnitEnum.DAYS).units(10L));
+
+      final ConnectionUpdate connectionUpdate = new ConnectionUpdate()
+          .connectionId(standardSync.getConnectionId())
+          .scheduleType(ConnectionScheduleType.BASIC) // update route requires this to be set even if it isn't changing
+          .scheduleData(newScheduleData);
+
+      final ConnectionRead expectedRead = ConnectionHelpers.generateExpectedConnectionRead(
+          standardSync.getConnectionId(),
+          standardSync.getSourceId(),
+          standardSync.getDestinationId(),
+          standardSync.getOperationIds(),
+          standardSync.getSourceCatalogId())
+          .schedule(new ConnectionSchedule().timeUnit(ConnectionSchedule.TimeUnitEnum.DAYS).units(10L)) // still dual-writing to legacy field
+          .scheduleType(ConnectionScheduleType.BASIC)
+          .scheduleData(newScheduleData);
+
+      final StandardSync expectedPersistedSync = Jsons.clone(standardSync)
+          .withSchedule(new Schedule().withTimeUnit(TimeUnit.DAYS).withUnits(10L)) // still dual-writing to legacy field
+          .withScheduleType(ScheduleType.BASIC_SCHEDULE)
+          .withScheduleData(new ScheduleData().withBasicSchedule(new BasicSchedule().withTimeUnit(BasicSchedule.TimeUnit.DAYS).withUnits(10L)))
+          .withManual(false);
+
+      when(configRepository.getStandardSync(standardSync.getConnectionId())).thenReturn(standardSync);
+
+      final ConnectionRead actualConnectionRead = connectionsHandler.updateConnection(connectionUpdate);
+
+      assertEquals(expectedRead, actualConnectionRead);
+      verify(configRepository).writeStandardSync(expectedPersistedSync);
+      verify(eventRunner).update(connectionUpdate.getConnectionId());
+    }
+
+    @Test
+    void testUpdateConnectionPatchAddingNewStream() throws Exception {
```

Comment:
curious, do the tests with the multiple streams test for stream sorting behaviour?

Reply:
I think they do implicitly (they were failing until I added the sorting behavior) but I like your suggestion of writing a test explicitly for the sort function, I'll add that in before merging